src/async-wrap.h

```diff
@@ -87,13 +87,22 @@ class AsyncWrap : public BaseObject {
     PROVIDERS_LENGTH,
   };
 
+  enum Flags {
```

Comment:
Since you’re passing it as ints, and since we have C++11 available: How about `enum class JSMethodFlags {` (which would implicitly have the underlying type `int`)?